tests/pipeline/controlflowanalysis/test_pattern_independent_restructuring.py

```diff
@@ -4851,7 +4851,7 @@ def test_head_is_no_loop_predecessor(task):
     assert isinstance(inner_break_cond := inner_body.children[7], ConditionNode)
 
     # block 8 cond:
-    assert block_8_cond.condition and str(task._ast.condition_map[block_8_cond.condition]) == "var_2 != 0x0"
+    assert block_8_cond.condition and str(task._ast.condition_map[~block_8_cond.condition]) == "var_2 == 0x0"
```

Comment:
kinda seems like you inverted a check here. Was it wrong before?

Reply:
This was changes in another PR, due to the error I fixed, I only changed it back. 

Without the above fix, we have the same value for two conditions that are not the same, because the SSA-values change. Therefore, we return the negated symbol.